lib/matplotlib/pyplot.py

```diff
@@ -2180,9 +2180,13 @@ def set_cmap(cmap):
     matplotlib.cm.register_cmap
     matplotlib.cm.get_cmap
     """
+    # The name of the colormap may be different than the registered name.
+    # We want the registered name for the rc setting
+    name = getattr(cmap, "name", cmap)
+    # Run the cmap (str or Colormap) through get_cmap to validate
+    # the cmap before updating the rc parameters
     cmap = get_cmap(cmap)
```

Comment:
Perhaps clearer to return a nonesense variable here?  Even with he comments, I thought the next line was a typo...

```suggestion
    _ = get_cmap(cmap)
```